neptune/internal/hardware/hardware_metric_reporting_job.py

```diff
@@ -0,0 +1,104 @@
+#
+# Copyright (c) 2020, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import os
+import time
+import uuid
+import socket
+
+from uuid import UUID
+
+from neptune_old.internal.hardware.metrics.metrics_container import MetricsContainer
+from neptune_old.internal.hardware.metrics.reports.metric_reporter import MetricReporter
+from neptune_old.internal.hardware.metrics.reports.metric_reporter_factory import MetricReporterFactory
+from neptune_old.internal.hardware.metrics.metrics_factory import MetricsFactory
+from neptune_old.internal.hardware.gauges.gauge_factory import GaugeFactory
+from neptune_old.internal.hardware.gpu.gpu_monitor import GPUMonitor
+from neptune_old.internal.hardware.resources.system_resource_info_factory import SystemResourceInfoFactory
+from neptune_old.internal.hardware.gauges.gauge_mode import GaugeMode
+from neptune_old.internal.hardware.system.system_monitor import SystemMonitor
+from neptune_old.utils import in_docker
+
+from neptune.internal.backends.neptune_backend import NeptuneBackend
+from neptune.internal.background_job import BackgroundJob
+from neptune.internal.threading.daemon import Daemon
+
+
+_logger = logging.getLogger(__name__)
+
+
+class HardwareMetricReportingJob(BackgroundJob):
+
+    def __init__(self, exp_uuid: UUID, backend: NeptuneBackend, period: float = 10):
+        self._exp_uuid = exp_uuid
+        self._backend = backend
+        self._period = period
+        self._random_execution_id = "{}-{}".format(socket.gethostname(), str(uuid.uuid4()))
+        self._thread = None
+        self._started = False
+
+    def start(self):
+        if not SystemMonitor.requirements_installed():
```

Comment:
Is there a reason why you dont set those variables in constructor?

Reply:
Just because they are needed only by a thraed and we do not want to create w thread if psutils is not installed.